runtime/lua/vim/treesitter/language.lua

```diff
@@ -6,8 +6,22 @@ local M = {}
 ---@type table<string,string>
 local ft_to_lang = {}
 
----@param filetype string
----@return string|nil
+--- Get the filetypes associated with {lang}.
+--- @param lang string Name of parser
+--- @return string[] filetypes
+function M.get_filetypes(lang)
```

Comment:
Can this be simplified with `vim.tbl_add_reverse_lookup`?

(Or maybe just create a second reverse lookup table in `register ` -- now that the tables are no longer directly exposed, that should be safe?)

Reply:
I really dislike `vim.tbl_add_reverse_lookup` for reasons and it makes typing very difficult.

We could even flip the mapping and have `lang_to_ft`. I actually only did it this way because it made the autocmd (now removed) simpler.

For now, I'd prefer to just have a single table. We can look at implementing a good bi-map implementation in the future. At least until we've decided on `'parser'`.

Reply:
Yeah, I don't think the reverse lookup in the same(!) table is a good idea, either. 

But if `'parser'` is still on the table, it'd make sense to have `parser_to_ft` be the primary table since that will be the lookup direction.

Reply:
`ft_to_parser` makes the following much easier (which will be added with `'parser'`:

```lua
a.nvim_create_autocmd('FileType', {
  group = a.nvim_create_augroup('vim.treesitter.language', {}),
  callback = function(args)
    if vim.bo[args.buf].parser ~= '' then
      return
    end

    local ft = vim.bo[args.buf].filetype
    if ft ~= '' and ft_to_lang[ft] then
      vim.bo[args.buf].parser = ft_to_lang[ft]
    end
  end
})
```

This will be executed much more often than `get_filetypes`.